src/app/Services/CheckService.ts

```diff
@@ -57,8 +78,18 @@ export class CheckService {
                     this.openWebPage(downloadAddress);
                 }
             });
+        } else if (this.checkSwCache()) {
+            this.updateServiceWorkerCache();
+            if (showAlert) {
+                Swal.fire({
+                    title: 'There is a new version of Kahla!',
+                    text: 'We are upgrading automatically at the background.\n You will be notified when done.',
```

Comment:
It seems that sending a new alert is too annoying. What about just display a yellow bar if the patch number is different. And show alert only when the major version number is different.

Reply:
This alert will only be shown when user manually click "check update" button.

Reply:
Ohh, that's fine.